generators/typescript/codegen/src/ast/Type.ts

```diff
@@ -18,8 +92,65 @@ export class Type extends AstNode {
             case "string":
                 writer.write("string");
                 break;
+            case "array":
+                writer.writeNode(this.internalType.value);
+                writer.write("[]");
+                break;
+            case "bool":
+                writer.write("boolean");
+                break;
+            case "date":
+                writer.write("Date");
+                break;
+            case "set":
+                writer.write("Set<");
+                writer.writeNode(this.internalType.value);
+                writer.write(">");
+                break;
+            case "number":
+                writer.write("number");
+                break;
+            case "record":
+                writer.write("Record<");
+                writer.writeNode(this.internalType.keyType);
+                writer.write(", ");
+                writer.writeNode(this.internalType.valueType);
+                writer.write(">");
+                break;
+            case "object":
+                writer.write("object");
+                break;
+            case "undefined":
+                writer.write("undefined");
+                break;
+            case "uuid":
+                writer.write("uuid");
+                break;
+            case "reference":
+                writer.writeNode(this.internalType.reference);
+                break;
+            case "literal":
+                if (typeof this.internalType.value === "string") {
+                    writer.write(`"${this.internalType.value}"`);
+                } else if (typeof this.internalType.value === "boolean") {
+                    writer.write(`${this.internalType.value}`);
+                }
+                break;
+            case "unknown":
+                writer.write("unknown");
+                break;
+            case "union": {
+                const numVariants = this.internalType.variants.length;
+                this.internalType.variants.forEach((variant, idx) => {
+                    writer.writeNode(variant);
+                    if (idx < numVariants - 1) {
+                        writer.write(" | ");
+                    }
+                });
```

Comment:
nit: similar to the other comment:
```suggestion
                this.internalType.variants.forEach((variant, idx) => {
                    if (idx > 0) {
                        writer.write(" | ");
                    }
                    writer.writeNode(variant);
                });
```